### ISSUE TITLE
Regex in value_transform

### ISSUE THREAD

**Issue by tchabit** [opsdis/aci-exporter#17]
Hello,

First of all, thank you for this great exporter !

I'm currently trying to get usage visibility on dynamic pool vlans, and in order to do so would need to be able to use as metric values the attributes 'to' and 'from' of class fvnsEncapBlk.
Problem is that those attributes values are in the form 'vlan-xxxx', so I need to remove the 'vlan-' part in order to have a float value.

I did not manage to find a way to use regex on the transform_value function, is it possible ?

BR

**Comment by thenodon** [comment]
Hi @tchabit and thanks for your feature request. Glad to hear that the aci-exporter is useful for you. Have had so few requests that I did not think anyone was using the exporter. 
I think your request is vary valid and I will try to fix it. It would be great if you can attach the `class_queries` configuration for your query.

**Comment by thenodon** [comment]
Is this the kind of output you are looking for?
```
# HELP vlans_from The from vlan
# TYPE vlans_from gauge
aci_vlans_from{aci="ACI Fabric1",fabric="cisco_sandbox",from="vlan-20",to="vlan-50",vlanns="production"} 20
aci_vlans_from{aci="ACI Fabric1",fabric="cisco_sandbox",from="vlan-1111",to="vlan-1111",vlanns="L3-Ext-vlan"} 1111
aci_vlans_from{aci="ACI Fabric1",fabric="cisco_sandbox",from="vlan-12",to="vlan-14",vlanns="Physical_vlpool"} 12
aci_vlans_from{aci="ACI Fabric1",fabric="cisco_sandbox",from="vlan-77",to="vlan-77",vlanns="L3_vlpool"} 77
# HELP vlans_to The to vlan
# TYPE vlans_to gauge
aci_vlans_to{aci="ACI Fabric1",fabric="cisco_sandbox",from="vlan-20",to="vlan-50",vlanns="production"} 50
aci_vlans_to{aci="ACI Fabric1",fabric="cisco_sandbox",from="vlan-1111",to="vlan-1111",vlanns="L3-Ext-vlan"} 1111
aci_vlans_to{aci="ACI Fabric1",fabric="cisco_sandbox",from="vlan-12",to="vlan-14",vlanns="Physical_vlpool"} 14
aci_vlans_to{aci="ACI Fabric1",fabric="cisco_sandbox",from="vlan-77",to="vlan-77",vlanns="L3_vlpool"} 77
# HELP scrape_duration_seconds The duration, in seconds, of the last scrape of the fabric
# TYPE scrape_duration_seconds gauge
aci_scrape_duration_seconds{aci="ACI Fabric1",fabric="cisco_sandbox"} 1.078294234
```

**Comment by tchabit** [comment]
Hello @thenodon,

Thank you for the quick response; I confirm that your exporter is used a lot here, in conjunction with a json exporter and python scripts for some API calls that we can't do FabricWide on our bigger Fabrics.

Regarding this request, the class_queries looks as followed, with fixed value_transform in order to be able through prometheus rules to calculate the block size and block utilization :
 
```
  vlanpool:
    class_name: fvnsEncapBlk
    query_parameter: '?query-target-filter=wcard(fvnsEncapBlk.dn,".*-dynamic*")'
    metrics:
      - name: vlanpool_block_first
        value_name: fvnsEncapBlk.attributes.from
        type: "gauge"
        help: "Returns the info of dynamic VLANs Block First Vlan"
        value_transform:
          'vlan-1000': 1000
          'vlan-1200': 1200
          'vlan-1400': 1400
          'vlan-1600': 1600
          'vlan-2100': 2100

      - name: vlanpool_block_last
        value_name: fvnsEncapBlk.attributes.to
        type: "gauge"
        help: "Returns the info of dynamic VLANs Block Last Vlan"
        value_transform:
          'vlan-1199': 1199
          'vlan-1399': 1399
          'vlan-1599': 1599
          'vlan-2099': 2099
          'vlan-2299': 2299

    labels:
      - property_name: fvnsEncapBlk.attributes.from
        regex: "^vlan-(?P<vlFirst>.*)"
      - property_name: fvnsEncapBlk.attributes.to
        regex: "^vlan-(?P<vlLast>.*)"
      - property_name: fvnsEncapBlk.attributes.dn
        regex: "^uni/infra/vlanns-\\[(?P<poolName>.*)\\]-dynamic/.*"
```

What would be great would be to be able to do something like that for the metrics part :

```
    metrics:
      - name: vlanpool_block_first
        value_name: fvnsEncapBlk.attributes.from
        type: "gauge"
        help: "Returns the info of dynamic VLANs Block First Vlan"
        value_transform_regexp:
          'vlan-(.*)': $1

      - name: vlanpool_block_last
        value_name: fvnsEncapBlk.attributes.to
        type: "gauge"
        help: "Returns the info of dynamic VLANs Block Last Vlan"
        value_transform_regexp:
          'vlan-(.*)': $1
```


The mock output you provide would be perfect ! Is it through a new function or an evolve of the current value_transform ?

**Comment by tchabit** [comment]
Seems great!

I confirm that we might use both `value_transform` and `value_transform_regexp` (you might have an `unknown` string value if deployment is faulty), therefore I'd execute the regexp form first aswell.

**Comment by thenodon** [comment]
@tchabit please try out branch [17-regex-in-value_transform](https://github.com/opsdis/aci-exporter/tree/17-regex-in-value_transform).
The transformation is called `value_regex_transformation`, so in your case set:
`value_regex_transformation: "vlan-(.*)"`
Please test it before its merge.

**Comment by thenodon** [comment]
@tchabit great to hear. Will merge. Do not to star the project :)

**Comment by thenodon** [comment]
Now merged to master and tagged as v0.3.0, latest release.

**Comment by tchabit** [comment]
Great, Thanks !
